cmd/soroban-cli/src/commands/contract/init.rs

```diff
@@ -105,656 +101,146 @@ struct Runner {
-                            path.to_string_lossy(),
-                            new_path
-                        ),
-                        e,
-                    )
-                })?;
-            }
-        }
+        Self::create_dir_all(project_path.join("contracts").as_path())?;
 
         Ok(())
     }
 
-    fn file_exists(file_path: &Path) -> bool {
-        metadata(file_path)
-            .as_ref()
-            .map(Metadata::is_file)
-            .unwrap_or(false)
-    }
-
-    fn check_internet_connection() -> bool {
-        if let Ok(_req) = get(GITHUB_URL).call() {
-            return true;
-        }
-
-        false
-    }
-
-    fn include_example_contracts(&self) -> bool {
-        !self.args.with_example.is_empty()
-    }
+    fn copy_template_files(&self) -> Result<(), Error> {
+        let binding = Path::new(&self.args.project_path)
+            .join("contracts")
+            .join(&self.args.name);
+        let project_path = binding.as_path();
 
-    fn clone_repo(from_url: &str, to_path: &Path) -> Result<(), Error> {
-        let mut prepare = clone::PrepareFetch::new(
-            from_url,
-            to_path,
-            create::Kind::WithWorktree,
-            create::Options {
-                destination_must_be_empty: false,
-                fs_capabilities: None,
-            },
-            open::Options::isolated(),
-        )
-        .map_err(|e| Error::PrepareFetch(Box::new(e)))?
-        .with_shallow(remote::fetch::Shallow::DepthAtRemote(
-            NonZeroU32::new(1).unwrap(),
+        self.print.infoln(format!(
+            "Adding package to the workspace at {project_path:?}"
         ));
 
-        let (mut checkout, _outcome) = prepare
-            .fetch_then_checkout(progress::Discard, &AtomicBool::new(false))
-            .map_err(|e| Error::Fetch(Box::new(e)))?;
-        let (_repo, _outcome) = checkout
-            .main_worktree(progress::Discard, &AtomicBool::new(false))
-            .map_err(Error::Checkout)?;
-
-        Ok(())
-    }
-
-    fn copy_example_contracts(
-        &self,
-        from: &Path,
-        to: &Path,
-        contracts: &[String],
-    ) -> Result<(), Error> {
-        let project_contracts_path = to.join("contracts");
-        for contract in contracts {
-            self.print
-                .infoln(format!("Initializing example contract: {contract}"));
-            let contract_as_string = contract.to_string();
-            let contract_path = Path::new(&contract_as_string);
-            let from_contract_path = from.join(contract_path);
-            let to_contract_path = project_contracts_path.join(contract_path);
-            Self::create_dir_all(&to_contract_path)?;
-
-            self.copy_contents(&from_contract_path, &to_contract_path)?;
-            Self::edit_contract_cargo_file(&to_contract_path)?;
+        if project_path.exists() {
+            return Err(AlreadyExists(self.args.name.clone()));
         }
```

Comment:
I think we should keep allowing users to overwrite a project. It's a great way to upgrade all your project, because you can undo undesired changes with git, while keeping everything else.

Reply:
The behavior has changed, so I don't think this flag is needed anymore